### PR TITLE
[ROCm] Update Dockerfiles of ROCm and MIGraphX to ROCm5.6

### PR DESCRIPTION
**Pull request by jeffdaily**
Update Dockerfiles of ROCm and MIGraphX to ROCm5.6.

CC @PeixuanZuo @cloudhan @TedThemistokleous @causten

### CODE REVIEW

File: tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM rocm/cupy:rocm5.5.0_ubuntu20.04_py3.8_pytorch2.0.0_cupy13.0.0
+FROM rocm/cupy:rocm5.6.0_ubuntu20.04_py3.8_pytorch2.0.0_cupy13.0.0
```
Comment (cloudhan) [inline tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile:1]: @PeixuanZuo I recall that you deliberately downgraded the image from 5.6 to 5.5 previous. What was the reason at that time?

Comment (PeixuanZuo) [reply tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile:1]: Need this fix(https://github.com/microsoft/onnxruntime/pull/16735) before update to ROCm5.6

File: tools/ci_build/github/linux/docker/migraphx-ci-pipeline-env.Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
-FROM rocm/pytorch:rocm5.5_ubuntu20.04_py3.8_pytorch_1.13.1
+FROM rocm/pytorch:rocm5.6_ubuntu20.04_py3.8_pytorch_1.13.1
```
Comment (PeixuanZuo) [inline tools/ci_build/github/linux/docker/migraphx-ci-pipeline-env.Dockerfile:1]: This Dockerfile is used for CI, we need to update more files to support it. You can refer to this PR(https://github.com/microsoft/onnxruntime/pull/15905) or let this Dockerfile unchanged, there is a draft PR to update CI dockerfile(https://github.com/microsoft/onnxruntime/pull/17076)

File: tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile (hunk 1 above)
Comment (PeixuanZuo) [reply tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile:1]: Same as this Dockerfile.

**Pull request closed without merging**